src/plugin/transform.ts

```diff
@@ -89,10 +92,17 @@ export function transform({
         mdDir.blockDirectives['html'] = plugin;
         mdDir.renderer.rules[TOKEN_TYPE] = (tokens, idx, _opts, _env, self) => {
             const token = tokens[idx];
-            const rendered = self.renderAttrs(token);
-            const resultRendered = sanitize ? sanitize(rendered) : rendered;
 
-            return `<${token.tag}${resultRendered}></${token.tag}>`;
+            if (sanitize && token.attrs) {
+                for (const [index, [attr, value]] of token.attrs.entries()) {
+                    if (attr === TokenAttr.srcdoc && token.attrs[index]) {
+                        token.attrs[index] = [attr, sanitize(value)];
+                        break;
+                    }
+                }
```

Comment:
you can use `token.attrGet()` and `token.attrSet()`, or `token.attrIndex()` methods, instead of iterating over `token.attrs` manually